crates/aptos/src/genesis/keys.rs

```diff
@@ -22,59 +23,92 @@ use serde::{Deserialize, Serialize};
 use std::path::PathBuf;
 
 const PRIVATE_KEYS_FILE: &str = "private-keys.yaml";
+const VALIDATOR_FILE: &str = "validator-identity.yaml";
+const VFN_FILE: &str = "validator-full-node-identity.yaml";
 
 /// Generate account key, consensus key, and network key for a validator
 #[derive(Parser)]
 pub struct GenerateKeys {
+    #[clap(flatten)]
+    prompt_options: PromptOptions,
     /// Output path for the three keys
     #[clap(long, parse(from_os_str), default_value = ".")]
     output_dir: PathBuf,
 }
 
 #[async_trait]
-impl CliCommand<PathBuf> for GenerateKeys {
+impl CliCommand<Vec<PathBuf>> for GenerateKeys {
     fn command_name(&self) -> &'static str {
         "GenerateKeys"
     }
 
-    async fn execute(self) -> CliTypedResult<PathBuf> {
-        let account_key = key::GenerateKey::generate_ed25519_in_memory();
-        let consensus_key = key::GenerateKey::generate_ed25519_in_memory();
-        // Start network key based off of the account key, we can update it later
-        let network_key =
-            x25519::PrivateKey::from_ed25519_private_bytes(&account_key.to_bytes()).unwrap();
+    async fn execute(self) -> CliTypedResult<Vec<PathBuf>> {
         let keys_file = self.output_dir.join(PRIVATE_KEYS_FILE);
+        let validator_file = self.output_dir.join(VALIDATOR_FILE);
+        let vfn_file = self.output_dir.join(VFN_FILE);
+        check_if_file_exists(keys_file.as_path(), self.prompt_options)?;
+        check_if_file_exists(validator_file.as_path(), self.prompt_options)?;
+        check_if_file_exists(vfn_file.as_path(), self.prompt_options)?;
+
+        let account_key = ConfigKey::new(key::GenerateKey::generate_ed25519_in_memory());
+        let consensus_key = ConfigKey::new(key::GenerateKey::generate_ed25519_in_memory());
+        let validator_network_key = ConfigKey::new(key::GenerateKey::generate_x25519_in_memory()?);
+        let full_node_network_key = ConfigKey::new(key::GenerateKey::generate_x25519_in_memory()?);
```

Comment:
let's clean up this ... generate in memory and use the traditional approach at a later date.